modules/common/react/lib/styles/focusRing.ts

```diff
@@ -80,3 +81,50 @@ export default function focusRing(
 
   return calculateFocusRing(argsToPass);
 }
+
+interface ThemedFocusRingOptions {
+  width?: number;
+  separation?: number;
+  animate?: boolean;
+  inset?: boolean;
+  innerColor?: string;
+  outerColor?: string;
+  memoize?: boolean;
+}
+
+/**
+ * A utility to create a canvas style focus ring around your widget.
+ * By default, this mixin will create a 2px focus ring tightly wrapped
+ * to the container (no whitespace).
+ *
+ * @returns {CSSObject} the css object for the focus ring style
+ */
+export function themedFocusRing(
```

Comment:
Interested to hear thoughts on this. I did this for a few reasons:

1. Unfortunately because we are usually calling this from within a class component, we can't use `React.useContext`. This means we need to pass the `theme` into the `focusRing` function. It would have been pretty nasty passing in every prop just to get `theme` at the end.

2. I'm often frustrated by the unclear nature of these props when you see them in use. The added complexity of an options object was worth it for the legibility and possibility for future changes. This now looks like
`themedFocusRing(theme, {separation: 2, animate: false})` instead of `focusRing(2, 2, false)`. I realize IDE hinting could help, but I still prefer the options object.

3. Rather than trying to hack the existing `focusRing` function to accept individual params vs. an options object without a breaking change, I figured a new function would be cleaner. We can consider deprecating `focusRing` in the future.

Reply:
I'm fine with this change. The `theme` context can be passed either through the Hooks API or the Context API. I think this function shouldn't be burdened to care _how_ `theme` gets here.